crates/matrix-sdk-ui/src/timeline/tests/event_filter.rs

```diff
@@ -200,3 +203,116 @@ async fn hide_failed_to_parse() {
 
     assert_eq!(timeline.inner.items().await.len(), 0);
 }
+
+#[async_test]
+async fn event_type_filter_include_only_room_names() {
+    // Only return room name events
+    let event_filter = TimelineEventTypeFilter::Include(vec![TimelineEventType::RoomName]);
+
+    let timeline = TestTimeline::new().with_settings(TimelineInnerSettings {
+        event_filter: Arc::new(move |event, _| event_filter.filter(event)),
+        ..Default::default()
+    });
+
+    // Add a non-encrypted message event
+    timeline
+        .handle_live_message_event(&ALICE, RoomMessageEventContent::text_plain("The first message"))
+        .await;
+    // And a couple of room name events
```

Comment:
Could we add another third kind of event, in this test and the other one, please? It will make it clear how the two filter kinds `Include` and `Exclude` are different.